src/utils/dates.js

```diff
@@ -10,9 +10,71 @@ const ONE_DAY_IN_MILLISECONDS = 86400000;
 export function getFutureDate(offset) {
 	return new Date(Date.now() + offset * ONE_DAY_IN_MILLISECONDS);
 }
-export function getDaysBetweenDates(previousPurchaseDate) {
-	const pastDate = previousPurchaseDate.toDate();
+export function calculateDaysDifferenceFromNow(dateToCompare) {
+	const comparisonDate = dateToCompare.toDate();
 	const presentDate = new Date();
-	const diffInMilliseconds = presentDate.getTime() - pastDate.getTime();
+	const diffInMilliseconds = presentDate.getTime() - comparisonDate.getTime();
 	return Math.round(diffInMilliseconds / ONE_DAY_IN_MILLISECONDS);
 }
+
+export function comparePurchaseUrgency(list) {
+	const inactive = [];
+	const overdue = [];
+	const future = [];
+	//iterate through the list and categorize each item
+	list.forEach((item) => {
+		//positive numbers represent the past, negative numbers represent the future
+		const days = calculateDaysDifferenceFromNow(item.dateNextPurchased);
+		if (days >= 60) {
+			/*
+				-If sixty or more days have passed since the last purchase, we consider the item inactive.
+				-We flip the days to negative* to represent inactivity because inactive items should be sorted in reverse order from overdue and future items. 
```

Comment:
This is a good type of comment to have. "Here's *why* we wrote it this way." Having that context will be helpful when Future You has to come back and figure out what this code is doing! Nice!